src/unitxt/task.py

```diff
@@ -10,42 +13,145 @@ class Tasker:
+
+            if (
+                prediction_type == metric_prediction_type
+                or prediction_type == Any
+                or metric_prediction_type == Any
+            ):
+                continue
+
+            raise ValueError(
+                f"The task's prediction type ({prediction_type}) and '{metric_name}' "
+                f"metric's prediction type ({metric_prediction_type}) are different."
+            )
+
+    def get_input_value(self, instance: Dict[str, Any], key: str) -> Any:
         try:
-            inputs = {key: instance[key] for key in self.inputs}
+            return instance[key]
         except KeyError as e:
             raise KeyError(
-                f"Unexpected FormTask input column names ({[key for key in self.inputs if key not in instance]})."
-                f"The available input names: {list(instance.keys())}"
+                f"Unexpected Task 'inputs' names: {[k for k in self.inputs if k not in instance]}."
+                f"The available names: {list(instance.keys())}."
             ) from e
+
+    def get_output_value(self, instance: Dict[str, Any], key: str) -> Any:
         try:
-            outputs = {key: instance[key] for key in self.outputs}
+            return instance[key]
         except KeyError as e:
             raise KeyError(
-                f"Unexpected FormTask output column names: {[key for key in self.outputs if key not in instance]}"
-                f" \n available names:{list(instance.keys())}\n given output names:{self.outputs}"
+                f"Unexpected Task 'outputs' names: {[k for k in self.outputs if k not in instance]}."
+                f"The available names: {list(instance.keys())}."
             ) from e
 
+    def check_instance_value_type(
+        self,
+        value: Any,
+        data_type: str,
+        field_name: str,
+        io_type: Literal["inputs", "outputs"],
+    ) -> Any:
+        data_type = self.get_prediction_type(data_type)
+
+        if isoftype(value, data_type):
+            return value
+
+        raise ValueError(
+            f"Passed {io_type} value '{value}' of field {field_name} is not of required "
+            f"type ({data_type})."
+        )
+
+    def process(
+        self, instance: Dict[str, Any], stream_name: Optional[str] = None
+    ) -> Dict[str, Any]:
+        self.check_metrics_type()
+
+        inputs = {
+            key: self.check_instance_value_type(
+                self.get_input_value(
+                    instance,
+                    key,
+                ),
+                data_type,
+                key,
+                "inputs",
+            )
+            for key, data_type in self.inputs.items()
+        }
+        outputs = {
+            key: self.check_instance_value_type(
+                self.get_output_value(instance, key),
+                data_type,
+                key,
+                "outputs",
+            )
+            for key, data_type in self.outputs.items()
+        }
```

Comment:
Instead of implementing this logic twice, i suggest to add a function to `type_utils.py`:
```python
def verify_required_schema(required_schema_dict, input_dict):
    ...
```
Reasons: (1) It should be tested as an independent unit. (2) In the near future it will be used for other operators (to possibly verify both their inputs and outputs).